python/lsst/rapid/analysis/butlerUtils.py

```diff
@@ -0,0 +1,682 @@
+# This file is part of rapid_analysis.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import lsst.daf.butler as dafButler
+import itertools
+import copy
+
+
+__all__ = ["makeDefaultLatissButler",
+           "sanitize_day_obs",
+           "getMostRecentDayObs",
+           "getSeqNumsForDayObs",
+           "getMostRecentDataId",
+           "getDatasetRefForDataId",
+           "getDayObs",
+           "getSeqNum",
+           "getExpId",
+           "datasetExists",
+           "sortRecordsByAttribute",
+           "getDaysOnSky",
+           "getExpIdFromDayObsSeqNum",
+           "updateDataIdOrDataCord",
+           "fillDataId",
+           "getExpRecordFromDataId",
+           "getDayObsSeqNumFromExposureId",
+           "removeDataProduct",
+           "getLatissOnSkyDataIds",
+           "_repoDirToLocation"]
+
+LATISS_DEFAULT_COLLECTIONS = ['LATISS/raw/all', 'LATISS/calib', "LATISS/runs/quickLook"]
```

Comment:
It's certainly a thought, but I'm not sure that's something I want to do here and now. I _think_ it would be OK, but I need to think about it a bit (and probably understand collections a bit better too really).